### PR TITLE
Add `json:",inline"` to all embedded resource fields

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

#558 identified that our OpenAPI schema was missing some spec fields because we were missing this tag. We fixed this for the embedded `ResourceSpec` type, but it's also relevant to the embedded `ResourceClaimSpec`, `ResourceStatus`, `ConditionedStatus`, and `BindingStatus` types.

This PR also effectively reverts #325. I still think it would be ideal to represent `BindingState` as an int with a sane zero value that marshaled to a JSON string, but it is currently impossible to override the type of the field that is used when generating an OpenAPI spec per kubernetes-sigs/controller-tools#155. Until that issue is closed it seems better to simply make this a string with a meaningless zero value.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

**Comment by negz** [comment]
I've tested this by creating a `RedisCluster` claim, ensuring it can provision a `CloudMemorystoreInstance`, and that the binding phase and conditions work as expected on both resources.

**Comment by jbw976** [comment]
Does the revert of #325 have any documentation impact? Do we need to update any user facing guides because of that change?

Should we open a new issue on our side to track adding a default binding state back in?

File: pkg/apis/core/v1alpha1/bindingphase.go [hunk 1]
```diff
@@ -14,60 +14,33 @@ See the License for the specific language governing permissions and
...
 
 	// BindingPhaseBound resources are bound to another resource.
-	BindingPhaseBound
+	BindingPhaseBound BindingPhase = "Bound"
 )
 
 // A BindingStatus represents the bindability and binding of a resource.
 type BindingStatus struct {
 	// Phase represents the binding phase of the resource.
+	// +kubebuilder:validation:Enum=Unbindable,Unbound,Bound
```
Comment (jbw976) [inline pkg/apis/core/v1alpha1/bindingphase.go]: is this validation saying that `BindingPhaseUnknown` is not a valid value in the api server? Does this mean that any client that creates a BindingStatus must ensure it is set to one of these 3 values before doing a `POST` to the api server?

**Comment by negz** [comment]
> Does the revert of #325 have any documentation impact? Do we need to update any user facing guides because of that change?

I don't believe so. It should be functionally identical from the user perspective.

Should we open a new issue on our side to track adding a default binding state back in?
Raised https://github.com/crossplaneio/crossplane/issues/566.

File: pkg/apis/core/v1alpha1/bindingphase.go (hunk 1 above)
Comment (negz) [reply pkg/apis/core/v1alpha1/bindingphase.go]: It is. No user or automation should be setting `BindingPhaseUnknown` once it has transitioned to another status. `BindingPhaseUnknown` should perhaps be `BindingPhaseUnset`.

**Pull request merged**